src/qibocal/protocols/two_qubit_interaction/mermin/protocol.py

```diff
@@ -0,0 +1,259 @@
+        default_factory=dict
+    )
+    """Mitigation matrix computed using the readout_mitigation_matrix protocol."""
+
+    @property
+    def targets(self):
+        return list(self.data.keys())
+
+
+@dataclass
+class MerminResults(Results):
+    """Mermin Results class."""
+
+    mermin: dict[tuple[QubitId, ...], npt.NDArray[np.float64]] = field(
+        default_factory=dict
+    )
+    """Raw Mermin value."""
+
+    mermin_mitigated: dict[tuple[QubitId, ...], npt.NDArray[np.float64]] = field(
+        default_factory=dict
+    )
+    """Mitigated Mermin value."""
+
+
+def _acquisition(
+    params: MerminParameters,
+    platform: Platform,
+    targets: list[list[QubitId]],
+) -> MerminData:
+    r"""Data acquisition for Mermin protocol using pulse sequences."""
+
+    thetas = np.linspace(0, 2 * np.pi, params.ntheta)
+    data = MerminData(thetas=thetas.tolist())
+    if params.apply_error_mitigation:
+        mitigation_data, _ = readout_mitigation_matrix.acquisition(
+            readout_mitigation_matrix.parameters_type.load(
+                dict(pulses=True, nshots=params.nshots)
+            ),
+            platform,
+            targets,
+        )
+
+        mitigation_results, _ = readout_mitigation_matrix.fit(mitigation_data)
+        data.mitigation_matrix = mitigation_results.readout_mitigation_matrix
+
+    for qubits in targets:
+        mermin_polynomial = get_mermin_polynomial(len(qubits))
+        readout_basis = get_readout_basis(mermin_polynomial)
+
+        for theta in thetas:
+            mermin_sequences = create_mermin_sequences(
+                platform, qubits, readout_basis=readout_basis, theta=theta
+            )
+            options = ExecutionParameters(nshots=params.nshots)
+            # TODO: use unrolling
+            for basis, sequence in mermin_sequences.items():
+                results = platform.execute_pulse_sequence(sequence, options=options)
+                frequencies = calculate_frequencies(results, qubits)
+                for state, frequency in enumerate(frequencies.values()):
+                    data.register_qubit(
+                        MerminType,
+                        tuple(qubits),
+                        dict(
+                            theta=np.array([theta]),
+                            basis=np.array([basis]),
+                            state=np.array([str(format(state, f"0{len(qubits)}b"))]),
+                            frequency=np.array([frequency]),
+                        ),
+                    )
+    return data
+
+
+def _fit(data: MerminData) -> MerminResults:
+    """Fitting for Mermin protocol."""
+    targets = data.targets
+    results = {qubits: [] for qubits in targets}
+    mitigated_results = {qubits: [] for qubits in targets}
+    mermin_polynomial = get_mermin_polynomial(len(targets))
+    basis = np.unique(data.data[targets[0]].basis)
+    mermin_coefficients = get_mermin_coefficients(mermin_polynomial)
+    for qubits in targets:
```

Comment:
```suggestion
    basis = np.unique(data.data[targets[0]].basis)
    for qubits in targets:
        mermin_polynomial = get_mermin_polynomial(len(qubits))
        mermin_coefficients = get_mermin_coefficients(mermin_polynomial)
```

I think this should fix Stavros' problems. Done to mirror the `acquisition` routine.